src/main.rs

```diff
@@ -40,11 +41,18 @@ fn main() -> Result<(), Error> {
 
             println!("{:?}", action);
         }
-
         /*
         TODO: Your implementation of a synthesizer should be here.
         Start with returning an oscillating wave determined by the `time`-variable
         */
+        // added the recommended code and changed it until it stopped giving errors but it is like not exactly working either
+        //by not exactly working i mostly mean bc the audio / ui isnt intergrated yet i cant see/hear it
+        // i aksi dont know the implications of adding a defailt previous phase value but it had to initialize to something in order to compile. also had to define some stuff
+        let mut previous_phase = 1.0;
```

Comment:
yes indeed there does need to be an initialized previous_phase. The previous phase is a bit of "state" which needs to hang around and get re-used each time this "synth" block loops. There are a few ways to do that

1. declare and initialize it in a higher scope, like in "main". This makes previous_phase a global and if there are lots of synths they all will have the same phase.
2. if your synth is a class, then previous_phase can be an attribute or instance variable (I'll have to learn Rust object lingo, I haven't gotten that far yet) - then each synth you create will have its own phase
3. Or if synth is a class, then previous_phase can be a class variable, which is like a global but scoped to the class. Then all instances of that class of synth would share the same phase.
4. There's also ways within a module to make thread-local globals or thread-shared globals but I'd save those for cases where needed. I'm not there yet... 

For (1) if you have more than one state variable and the caller manages state, you'd have a struct that has all the state in it, so all the caller needs is one variable for all the state.

I think we don't have a synth class here so use approach (1) declare previous_phase near the top of main() and then the synth loop can just use it. Later you might pass a pointer to previous_phase in and share it.